src/Visualization/Python/PlotPowerMonitors.py

```diff
@@ -0,0 +1,312 @@
+                all_modes[d] += modes_dim**2
+        for d in range(element.dim):
+            all_modes[d] = np.sqrt(all_modes[d])
+            # Collect reduction data
+            # - We estimate the truncation error by just taking the highest
+            #   mode. This won't work well with filtering and should be improved
+            #   on the C++ side.
+            max_error[subplot_index][d] = max(max_error[subplot_index][d],
+                                              all_modes[d][-1])
+        # Plot
+        for d, modes_dim in enumerate(all_modes):
+            ax.semilogy(modes_dim, **props_dim[d], zorder=30 + d)
+            ax.scatter(len(modes_dim) - 1,
+                       modes_dim[-1],
+                       marker=".",
+                       color=props_dim[d].get("color", "black"),
+                       zorder=30 + d)
+
+    for subplot_index, ax in enumerate(axes[0]):
+        ax.set_title(block_or_group_names[subplot_index], loc="left")
+        ax.set_title(f"{num_elements[subplot_index]} element" +
+                     "s"[:num_elements[subplot_index] != 1],
+                     loc="right")
+        ax.grid(which='both', zorder=0)
+        # Annotate the max truncation error. Also serves as a legend.
+        for d in range(domain.dim):
+            ax.axhline(max_error[subplot_index][d],
+                       **props_dim[d],
+                       zorder=20 + d)
+            ax.annotate(["$x$", "$y$", "$z$"][d],
+                        xy=(0, max_error[subplot_index][d]),
+                        xytext=((2 * d + 0.5) * plt.rcParams["font.size"], 0),
+                        textcoords='offset points',
+                        ha='left',
+                        va='center',
+                        bbox=dict(fc="white",
+                                  ec=props_dim[d].get("color", "black"),
+                                  pad=2.),
+                        zorder=40 + d)
+
+    # Add x-label spanning all subplots
+    ax_colspan = fig.add_subplot(111, frameon=False)
+    ax_colspan.tick_params(labelcolor='none',
+                           top=False,
+                           bottom=False,
+                           left=False,
+                           right=False)
+    ax_colspan.grid(False)
+    ax_colspan.set_xlabel("Mode number")
+
+
+@click.command()
+@click.argument("h5_files",
+                nargs=-1,
+                type=click.Path(exists=True,
+                                file_okay=True,
+                                dir_okay=False,
+                                readable=True))
+@click.option("--subfile-name",
+              "-d",
+              help=("Name of volume data subfile within the h5 files."))
+@click.option("--step",
+              type=int,
+              help=("Observation step number. "
+                    "Specify '-1' for the last step in the file. "
+                    "Mutually exclusive with '--time'."))
```

Comment:
How hard would it be to use `last` or `Last` to specify the last step rather than `-1`. When I was looking at the command, the `-1` looked like its own option unrelated to `--step` (because of the hyphen/negative).

Reply:
done